src/pykeen/nn/modules.py

```diff
@@ -2549,17 +2550,70 @@ def forward(
         )
 
 
+class ClampedInteraction(Interaction[HeadRepresentation, RelationRepresentation, TailRepresentation]):
+    """An adapter to clamp scores to a minimum or maximum value.
+
+    .. warning::
+        The used :func:`torch.clamp` function has zero gradient for scores below the minimum of above the maximum value.
+        Thus, it aggravates gradient-based optimization.
+    """
+
+    # TODO: this seems to cause cyclic imports during sphinx
+    # @update_docstring_with_resolver_keys(ResolverKey(name="base", resolver="pykeen.nn.modules.interaction_resolver"))
+    def __init__(
+        self,
+        clamp_score: None | float | tuple[float | None, float | None] = None,
+        base: HintOrType[Interaction[FloatTensor, FloatTensor, FloatTensor]] = DistMultInteraction,
```

Comment:
should this be more generic and reuse `Interaction[HeadRepresentation, RelationRepresentation, TailRepresentation]`? in theory, couldn't you clamp any kind of interaction?

Reply:
done in https://github.com/pykeen/pykeen/pull/1470/commits/6df048e74db1ac68f842cc67c217fe32b889a64b